app/status.go

```diff
@@ -282,6 +284,112 @@ func (a *App) SetStatusAwayIfNeeded(userID string, manual bool) {
 	a.SaveAndBroadcastStatus(status)
 }
 
+func (a *App) SetStatusDoNotDisturbSchedule(userId, currentDayOfTheWeek, currentTime string) {
```

Comment:
how will status be updated? are you polling from client periodically?

Reply:
> how will status be updated? are you polling from client periodically?

You're right . To get user's schedules periodically, we save user's working hours in the DB according to the user's day and status will be updated by calling the user's schedules.